src/usePlanshipCustomer.ts

```diff
@@ -1,9 +1,26 @@
 import { useContext } from 'react'
 
 import context, { IPlanshipCustomerContext } from './customerContext'
+import { CustomerSubscriptionWithPlan, Entitlements, PlanshipCustomer } from '@planship/fetch'
+import { EntitlementsBase } from './types'
 
-const usePlanshipCustomer = () => {
-  return useContext<IPlanshipCustomerContext>(context)
+interface ICustomerContext<T extends EntitlementsBase> {
+  planshipCustomerApiClient?: PlanshipCustomer
+  entitlements: T
+  subscriptions: CustomerSubscriptionWithPlan[]
 }
 
-export { usePlanshipCustomer }
+export function usePlanshipCustomer(): IPlanshipCustomerContext
+export function usePlanshipCustomer<TEntititlements extends EntitlementsBase>(entitlementsType: {
+  new (entitlementsDict: Entitlements): TEntititlements
+}): ICustomerContext<TEntititlements>
+export function usePlanshipCustomer(entitlementsType?: { new (entitlementsDict: Entitlements): object }) {
+  const currentContext = useContext<IPlanshipCustomerContext>(context)
+  if (entitlementsType)
```

Comment:
Not sure if we're enforcing braces around if logic...

Reply:
Yeah, just thinking we should be consistent. No strong opinion but if the logic is on a separate line than the subsequent code it seems like braces help make it clearer.